src/TraceEvent/TraceEventSession.cs

```diff
@@ -1116,6 +1116,22 @@ public void CaptureState(Guid providerGuid, ulong matchAnyKeywords = ulong.MaxVa
                     asArray[6] = (byte)(longVal >> 48);
                     asArray[7] = (byte)(longVal >> 56);
                 }
+                // Query existing keywords and merge them with requested keywords before capture state
+                ulong mergedKeywords = matchAnyKeywords;
+                Guid* providerPtr = stackalloc Guid[1];
+                *providerPtr = providerGuid;
+                long? existingKeywords = GetEnabledKeywordsForProviderAndSession(providerPtr, (ulong)m_SessionId);
```

Comment:
We will also need the level.  Please modify `GetEnabledKeywordsForProviderAndSession` instead return a struct containing the keywords and level.  You can rename it `GetEnabledInfoForProviderAndSession`.  The level in the setruct should be of type `TraceEventLevel`.  Then, pass the level to the call to `EnableTraceEx2` that enables the provider.  Leave the call to CAPTURE_STATE with a `Verbose` level.